sunpy/net/jsoc/jsoc.py

```diff
@@ -650,27 +645,19 @@ def _make_recordset(self, series, start_time='', end_time='', wavelength='',
                                  "prime-keys is not yet supported. Alternative is to "
                                  "use only one of the primekey to query for series data.")
                 raise ValueError(error_message)
-            if match:
```

Comment:
As valid as these logic refactors are doing them in the middle of this PR which is fixing a legitimate bug, in code that most of us are not familiar with makes reviewing this PR *really* hard.

Reply:
Don't worry about it.

Reply:
Well if you care about faster reviews then it might be something to think about :wink:

Reply:
*I won't reply to that with my actual thoughts.*

But yes, that is very true. I will endeavour to make sure my PR's are smaller in the future.